src/collection-view.js

```diff
@@ -101,8 +101,9 @@ Marionette.CollectionView = Marionette.View.extend({
   _onCollectionAdd: function(child, collection, opts) {
     var index;
     if (opts.at !== undefined && !this.getOption('filter')) {
-      index = opts.index;
-      if (!index) {
+      if (typeof opts.index === 'number') {
+        index = opts.index;
```

Comment:
This line still needs coverage--have to look deeper into Backbone and where `index` is coming from.